resolver_mac.c

```diff
@@ -33,78 +33,89 @@ typedef struct proxy_resolver_mac_s {
     char *list;
 } proxy_resolver_mac_s;
 
+static const char *proxy_resolver_mac_get_proxy_scheme(CFStringRef proxy_type) {
+    const char *scheme = NULL;
+    if (CFEqual(proxy_type, kCFProxyTypeNone))
+        scheme = "direct://";
+    else if (CFEqual(proxy_type, kCFProxyTypeHTTP))
+        scheme = "http://";
+    else if (CFEqual(proxy_type, kCFProxyTypeHTTPS))
+        // "HTTPS Proxy" on macOS means "use CONNENCT verb for a https:// URL", the proxy still should be an HTTP proxy.
```

Comment:
So this seems to suggest that macOS does not support `HTTPS` directive from PAC file. However, an HTTPS proxy can still probably be configured in settings dialog?

https://support.apple.com/guide/mac-help/change-proxy-settings-on-mac-mchlp2591/mac
![image](https://github.com/nmoinvaz/proxyres/assets/1364432/6761efc9-67c3-4938-8233-8555c30b347a)

Is that your take as well?

Reply:
IDK, it is first time mentioned in macOS Montgomery, but we are targeting macOS 10.13. It may change the behavior if we change the target. If it still returns `CFProxyTypeHTTPS` for the `PROXY` directive, it is useless for us.